src/v/storage/segment_utils.cc

```diff
@@ -377,66 +377,63 @@ ss::future<> do_compact_segment_index(
           return write_clean_compacted_index(reader, cfg, resources);
       });
 }
+
 ss::future<storage::index_state> do_copy_segment_data(
```

Comment:
nit: Should we make this `noexcept`?

Reply:
good question. yes? but noexcept just calls std::terminate if an exception is raised in the body. then again, this function is not equipped to handle stray exceptions and they will bubble up to other places that might not be equipped, so the end result would just be a messier abort. i think I'll just have a separate pr for this, and keep an eye on the ci for crashes after this coroutine conversion

Reply:
> then again, this function is not equipped to handle stray exceptions

Yeah, that's why I suggested it.